scripts/test_solidity_lsp.py

```diff
@@ -0,0 +1,342 @@
+            "diagnostic: check range"
+        )
+    # }}}
+
+    # {{{ actual tests
+    def test_publish_diagnostics_1(self, solc: JsonRpcProcess) -> None:
+        """ Publish diagnostics (warnings) """
+        TEST_NAME = 'publish_diagnostics_1'
+        published_diagnostics = self.open_file_and_wait_for_diagnostics(solc, TEST_NAME)
+
+        self.expect_equal(len(published_diagnostics), 1, "One published_diagnostics message")
+        report = published_diagnostics[0]
+
+        self.expect_equal(report['uri'], self.get_test_file_uri(TEST_NAME), "Correct file URI")
+        diagnostics = report['diagnostics']
+
+        self.expect_equal(len(diagnostics), 3, "3 diagnostic messages")
+        self.expect_diagnostic(diagnostics[0], code=6321, lineNo=13, startColumn=44, endColumn=48)
+        self.expect_diagnostic(diagnostics[1], code=2072, lineNo= 7, startColumn= 8, endColumn=19)
+        self.expect_diagnostic(diagnostics[2], code=2072, lineNo=15, startColumn= 8, endColumn=20)
+
+    def test_publish_diagnostics_2(self, solc: JsonRpcProcess) -> None:
+        """ Publish diagnostics (errors) """
+        TEST_NAME = 'publish_diagnostics_2'
+        published_diagnostics = self.open_file_and_wait_for_diagnostics(solc, TEST_NAME)
+
+        self.expect_equal(len(published_diagnostics), 1, "One published_diagnostics message")
+        report = published_diagnostics[0]
+
+        self.expect_equal(report['uri'], self.get_test_file_uri(TEST_NAME), "Correct file URI")
+        diagnostics = report['diagnostics']
+
+        self.expect_equal(len(diagnostics), 3, "3 diagnostic messages")
+        self.expect_diagnostic(diagnostics[0], code=9574, lineNo= 7, startColumn= 8, endColumn=21)
+        self.expect_diagnostic(diagnostics[1], code=6777, lineNo= 8, startColumn= 8, endColumn=15)
+        self.expect_diagnostic(diagnostics[2], code=6160, lineNo=18, startColumn=15, endColumn=36)
+
+    def test_didOpen_with_relative_import(self, solc: JsonRpcProcess) -> None:
+        """ didOpen with relative import """
+        TEST_NAME = 'didOpen_with_import'
+        published_diagnostics = self.open_file_and_wait_for_diagnostics(solc, TEST_NAME, 2)
+
+        self.expect_equal(len(published_diagnostics), 2, "Diagnostic reports for 2 files")
+
+        # primary file:
+        report = published_diagnostics[0]
+        self.expect_equal(report['uri'], self.get_test_file_uri(TEST_NAME), "Correct file URI")
+        self.expect_equal(len(report['diagnostics']), 0, "one diagnostic")
+
+        # imported file (./lib.sol):
+        report = published_diagnostics[1]
+        self.expect_equal(report['uri'], self.get_test_file_uri('lib'), "Correct file URI")
+        self.expect_equal(len(report['diagnostics']), 1, "one diagnostic")
+        self.expect_diagnostic(report['diagnostics'][0], code=2072, lineNo=12, startColumn=8, endColumn=19)
+    # }}}
+    # }}}
+
+if __name__ == "__main__":
+    suite = SolidityLSPTestSuite()
+    rv = suite.main()
```

Comment:
Does `rv` mean "return value"?
```suggestion
    exit_code = suite.main()
```

Reply:
habbits, yes. changed. :)